the-collab-lab/tcl-20-whats-near-me
# user clicks on the pin & info window is displayed

## Description
- User clicks on the pin and info window appears with a larger picture (if available), name, description & link to wikipedia
- User clicks the x on the info window to close
- Added react context api (in the context folder & LocationsContext.js file) to centralize app architecture 



## Related Issue
Story: Integrate with "near me" API

## Acceptance Criteria
- [x] User can tap on a pin on the map to open an Info Window with the following details about the place:
- Name
- Description
- Thumbnail (if there is one) for the place rendered at the largest size possible within the Info Window
- Distance of the place from the map center, in kms

- [x] A link to view the place on Wikipedia
-Links out to Wikipedia should use the pageid property that is returned by the API, using the following format: https://en.wikipedia.org/?curid=<pageid> (e.g., https://en.wikipedia.org/?curid=60401407)


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓  | :sparkles: New feature     |
| ✓  | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before
![image](https://user-images.githubusercontent.com/59700665/105269021-2bd0ed80-5b61-11eb-837a-7b49773652be.png)


### After
When you click on the icon an info window pops up! 
![image](https://user-images.githubusercontent.com/59700665/105269103-4c00ac80-5b61-11eb-8a6b-be91c087b2e4.png)

The wikipedia link will open the page in a new tab with corresponding url.
![image](https://user-images.githubusercontent.com/59700665/105269121-54f17e00-5b61-11eb-9058-e0f783879500.png)

An example of an info window with a thumbnail picture. 
![image](https://user-images.githubusercontent.com/59700665/105269074-430fdb00-5b61-11eb-947b-e9919bbcffbd.png)


## Testing Steps / QA Criteria

Given a user viewing the map view of the app with pins rendered, when the user taps on a pin, then an Info Window should open showing more details about the place.

A user should be able click on the wikipedia link and a new tab should open to the corresponding page. It will link out to Wikipedia should use the pageid property that is returned by the API, using the following format: https://en.wikipedia.org/?curid=<pageid> (e.g., https://en.wikipedia.org/?curid=60401407)

Given a user viewing an Info Window with more details, when they tap the close “x” the Info Window should close, returning them to the view before they opened the Info Window.
